test/e2e/atlas/cleanup_test.go

```diff
@@ -65,6 +65,9 @@ func TestCleanup(t *testing.T) {
 				t.Parallel()
 				deleteAllDataFederations(t, cliPath, projectID)
 			})
+			t.Cleanup(func() {
```

Comment:
why clean up and not a normal run?

Reply:
I decided to keep at clean up so that it runs last, to avoid deleting the keys before time when we would actually want to use them?

Reply:
merging but happy to follow-up on moving out of cleanup if we there are any concerns!